adafruit_display_text/scrolling_label.py

```diff
@@ -26,36 +26,42 @@
 __version__ = "0.0.0-auto.0"
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_Display_Text.git"
 
+try:
+    from typing import Union, Optional
+    from fontio import BuiltinFont
+    from adafruit_bitmap_font.bdf import BDF
+    from adafruit_bitmap_font.pcf import PCF
+except ImportError:
+    pass
+
 import time
 from adafruit_display_text import bitmap_label
 
 
 class ScrollingLabel(bitmap_label.Label):
-
-    """
-    ScrollingLabel - A fixed-width label that will scroll to the left
+    """ScrollingLabel - A fixed-width label that will scroll to the left
     in order to show the full text if it's larger than the fixed-width.
 
     :param font: The font to use for the label.
-    :param max_characters: The number of characters that sets the  fixed-width. Default is 10.
-    :param text: The full text to show in the label. If this is longer than
-     `max_characters` then the label will scroll to show everything.
-    :param animate_time: The number of seconds in between scrolling animation
+    :type font: ~BuiltinFont, ~BDF, or ~PCF
+    :param int max_characters: The number of characters that sets the fixed-width. Default is 10.
+    :param str text: The full text to show in the label. If this is longer than
+     ``max_characters`` then the label will scroll to show everything.
+    :param float animate_time: The number of seconds in between scrolling animation
      frames. Default is 0.3 seconds.
-    :param current_index: The index of the first visible character in the label.
-     Default is 0, the first character. Will increase while scrolling.
-    """
+    :param int current_index: The index of the first visible character in the label.
+     Default is 0, the first character. Will increase while scrolling."""
 
     # pylint: disable=too-many-arguments
     def __init__(
         self,
-        font,
-        max_characters=10,
-        text="",
-        animate_time=0.3,
-        current_index=0,
+        font: Union[BuiltinFont, BDF, PCF],
+        max_characters: Optional[int] = 10,
```

Comment:
So `Optional` is actually a horrible name for what it describes :sweat_smile:  What it denotes is something that can take `None` as a value.  For example if I had an argument for a class that could set the color for something as a hex code, or `None` to signify that it should be transparent, it would be:

```python
def set_color(color: Optional[int]):
    ...
```

It looks like these can't take `None` as a value, so despite being optional to provide, they aren't `Optional`.

In this case, despite being optional to provide, the argument still is/expects to be a `str` type, so it would just be:

```suggestion
        text: str = "",
```

Same for the ones below!